test/system/users_test.rb

```diff
@@ -0,0 +1,119 @@
+    admin = users(:admin_user)
+    viewer_email = 'user@example.com'
+
+    #
+    # Administrator sends a user invitation
+    #
+    Capybara.using_session('admin') do
+      visit root_path
+
+      click_on "Login"
+
+      fill_in 'Email', with: admin.email
+      fill_in 'Password', with: 'testpassword'
+      click_on 'Log in'
+
+      assert page.has_content?("Logged in as #{admin.email}"), "Admin should have an active session"
+      click_on 'Admin'
+
+      perform_enqueued_jobs do
+        fill_in 'E-mail:', with: viewer_email
+        click_on 'Generate Invitation'
+      end
+    end
+
+    #
+    # User accepts the invitation and creates and confirms their account
+    #
+    Capybara.using_session('user') do
+      open_email(viewer_email)
+      current_email.find('a[href*="invitation"]').click
+      clear_emails
+
+      fill_in 'Password', with: 'testpassword'
+      fill_in 'Password confirmation', with: 'testpassword'
+      click_on 'Sign up'
+
+      open_email(viewer_email)
+      current_email.click_link 'Confirm my account'
+      clear_emails
+
+      fill_in 'Email', with: viewer_email
+      fill_in 'Password', with: 'testpassword'
+      click_on 'Log in'
+
+      assert page.has_content?("Logged in as #{viewer_email}"), "User should have an active session"
+    end
+
+    #
+    # Administrator promotes the User to have the "admin" permission
+    #
+    Capybara.using_session('admin') do
+      visit admin_path
+
+      user_row = page.all('tr').find { |tr| tr.has_content? viewer_email }
+      within user_row do
+        click_on 'Promote to admin'
+      end
+    end
+
+    #
+    # User verifies that they have received the "admin" permission
+    #
+    Capybara.using_session('user') do
+      visit root_path
+
+      assert page.has_link?('Admin'), "User should have admin permissions"
+    end
+
+    #
+    # Administrator removes the User's "admin" permission
+    #
+    Capybara.using_session('admin') do
+      visit admin_path
+
+      user_row = page.all('tr').find { |tr| tr.has_content? viewer_email }
+      within user_row do
+        click_on 'Demote from admin'
+      end
+    end
+
+    #
+    # User verifies that the admin permission has been removed
+    #
+    Capybara.using_session('user') do
+      visit root_path
+      refute page.has_link?('Admin'), "User should not have admin permissions"
+    end
+
+    #
+    # Administrator deletes the User's account
+    #
+    Capybara.using_session('admin') do
+      visit admin_path
+
+      user_row = page.all('tr').find { |tr| tr.has_content? viewer_email }
+      within user_row do
+        click_on 'Delete Account'
+      end
+    end
+
+    #
+    # User is logged out from the deleted account and prevented from logging back in
+    #
+    Capybara.using_session('user') do
+      visit root_path
+      refute page.has_content?("Logged in as #{viewer_email}"), "User should NOT have an active session"
+
+      fill_in 'Email', with: viewer_email
```

Comment:
This is failing for me now! I think this used to have `click_on 'Login'` before it.

On a separate note, I just realized that this test is not getting run in CI. Can you add another step for it in `.circleci/config.yml` after the existing test step? https://github.com/edgi-govdata-archiving/web-monitoring-db/blob/c2c3e57045c96b5a0adadd1e4d5b70dd08c98a73/.circleci/config.yml#L36-L41

I think the new step just needs to run `bundle exec rake test:system`, right?

Reply:
I went with `bin/rails test:system test` which is the magical order according to https://stackoverflow.com/a/49830043/241735

Reply:
Huh, wonder what it is about Rake or the CLI parsing that makes the order so sensitive. Anyway, TIL!